lockfile/src/cyclonedx.rs

```diff
@@ -164,23 +164,37 @@ fn from_purl<T: Component>(component: &T) -> anyhow::Result<Package> {
 
 pub struct CycloneDX;
 
+impl CycloneDX {
+    fn process_components<T: Component>(components: Option<&[T]>) -> anyhow::Result<Vec<Package>> {
```

Comment:
If this function is going to `.map_or(Ok(Vec::new()), _)` on `components`, `components` can be a `&[T]`. [`&[T]` implements `Default`](https://doc.rust-lang.org/std/primitive.slice.html#impl-Default-for-%26%5BT%5D) so you can simplify by using `.unwrap_or_default()` and skip this `map_or` stuff.